New Entry: General JSP (#881)
Codecademy/docs: content/general/concepts/jsp/jsp.md

---
Title: 'JSP'
Description: 'JSP (Jakarta Server Pages, formerly JavaServer Pages) is a technology that allows a web developer to mix textual content with Java code that is executed on the web server to render the final web page.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:

>   - 'Server-Side Rendering'

Reviewer: ```suggestion
  - 'Dynamic Programming'
  - 'Servers'
  - 'Server-Side Rendering'
```

Reviewer: If possible, let's have at least two tags.

CatalogContent:
  - 'learn-java'
  - 'paths/back-end-engineer-career-path'
---

**JSP** (Jakarta Server Pages, formerly JavaServer Pages) is a technology that allows a web developer to mix textual content with [Java](https://www.codecademy.com/resources/docs/java) code that is executed on the web server to render the final web page. In this way it is similar to [PHP](https://www.codecademy.com/resources/docs/php) except that it uses the Java programming language. With JSP the developer can create dynamic pages based on [HTML](https://www.codecademy.com/resources/docs/html), [XML](https://www.codecademy.com/resources/docs/general/xml), SOAP or other document types.

JSP was released in 1999 by Sun Microsystems, and version 3.1 was released on April 31, 2022.

> ## Advantages

Reviewer: No disadvantages?

JSP has the following advantages compared to other methods of creating dynamic web pages.

- The execution of JSP pages is faster than other technologies.

> - JSP pages are compiled before being processed by the server reducing the load on the server.

Reviewer: ```suggestion
- JSP pages are compiled before being processed by the server, reducing the load on the server.
```

Is it me, or does this read weirdly without a comma?

- JSP has access to all the Java APIs.

> - Since JSP uses the Java Virtual Machine, JSP is platform agnostic and portable.

Reviewer: ```suggestion
- Since JSP uses the Java Virtual Machine, JSP is platform-agnostic and portable.
```

---
Title: 'JSP'
Description: 'JSP (Jakarta Server Pages, formerly JavaServer Pages) is a technology that allows a web developer to mix textual content with Java code that is executed on the web server to render the final web page.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Dynamic Programming'
  - 'Servers'
  - 'Server-Side Rendering'
CatalogContent:
  - 'learn-java'
  - 'paths/back-end-engineer-career-path'
---

> **JSP** (Jakarta Server Pages, formerly JavaServer Pages) is a technology that allows a web developer to mix textual content with [Java](https://www.codecademy.com/resources/docs/java) code that is executed on the web server to render the final web page. In this way it is similar to [PHP](https://www.codecademy.com/resources/docs/php) except that it uses the Java programming language. With JSP the developer can create dynamic pages based on [HTML](https://www.codecademy.com/resources/docs/html), [XML](https://www.codecademy.com/resources/docs/general/xml), SOAP or other document types.

Reviewer: ```suggestion
**JSP** (Jakarta Server Pages, formerly JavaServer Pages) is a technology that allows a web developer to mix textual content with [Java](https://www.codecademy.com/resources/docs/java) code that is executed on the web [server](https://www.codecademy.com/resources/docs/general/server) to [render](https://www.codecademy.com/resources/docs/general/server-side-rendering) the final web page. In this way it is similar to [PHP](https://www.codecademy.com/resources/docs/php) except that it uses the Java programming language. With JSP the developer can create dynamic pages based on [HTML](https://www.codecademy.com/resources/docs/html), [XML](https://www.codecademy.com/resources/docs/general/xml), SOAP, or other document types.
```

JSP was released in 1999 by Sun Microsystems, and version 3.1 was released on April 31, 2022.

## Advantages & Disadvantages

> JSP has the following advantages compared to other methods of creating dynamic web pages.

Reviewer: ```suggestion
JSP has the following advantages compared to other methods of creating dynamic web pages:
```

- The execution of JSP pages is faster than other technologies.
- JSP pages are compiled before being processed by the server, reducing the load on the server.
- JSP has access to all the Java APIs.

> - Since JSP uses the Java Virtual Machine, JSP is platform-agnostic and portable.

Reviewer: ```suggestion
- Since JSP uses the Java Virtual Machine, it is portable and platform-agnostic.
```

These are some disadvantages of JSP:

- Debugging and tracing errors can be difficult.
- It uses roughly twice the disk storage as other dynamic web pages.

> - JSP file must be compled on the server when first accessed, which can cause a noticable delay on the first visit to a page.

Reviewer 2: ```suggestion
-  A JSP file must be compiled on the server when first accessed, which can cause a noticeable delay on the first visit to a page.
```
